weave/feedback.py

```diff
@@ -0,0 +1,241 @@
+import json
+from datetime import datetime
+from typing import Any, Collection, Iterable, Iterator, Optional
+
+from rich.console import Console
+from rich.table import Table
+
+from . import graph_client_context
+from . import util
+from weave import pydantic_util
+from weave.trace_server import trace_server_interface as tsi
+from weave.trace_server.interface.query import Query
+from weave.trace.refs import parse_uri
+
+
+class Feedbacks:
```

Comment:
I am seeing that we have 4 data structures:
1. `tsi.Feedback` -> The common `Feedback` data pydantic schema
2. `Feedbacks` -> A container of `tsi.Feedback`s 
   * used primarily for display
   * returned from `RefFeedback.query`
   * Implements an iterator protocol over `tsi.Feedback`
3. `RefFeedback` -> A "pointer" to feedback for a refable thing.
   * Implements an iterator protocol over `tsi.Feedback`
   * Lazily fetches data
   * Supports mutation operations (because it is relates to a specific "ref")
4. Implied: `Iterator[tsi.Feedback]`

I'm not sure exactly why, but `Feedbacks` and `RefFeedback` seem to overlap in an important way. 

Ok, i think either:
1. `RefFeedback` wants to inherit from `Feedbacks`
2. `RefFeedback` should contain a `Feedbacks` instead of `items`

For some reason i also feel like they are the same thing. `RefFeedback` and `Feedbacks` both can be thought of as `FeedbackQuery`. I think the only difference is that `RefFeedback` has the added property that you can add/purge. The other difference is that `Feedbacks` is pre-materialized, but i think you could also make `Feedbacks` lazy.

Ok, talking this out, consider changing to two classes: `FeedbackQuery` and `RefFeedbackQuery(FeedbackQuery)`. 

`FeedbackQuery` basically is merging of both `RefFeedback` and `Feedbacks` (except for `add/purge`). It is lazy, iterator, nicely prints, etc... It has an internal `_query` property holding the current query. It also has a `query` method that returns a new `FeedbackQuery` with the query append to the existing query.

Then, `RefFeedbackQuery` is a subclass with an additional `weave_ref` property. This is where add/purge goes. 

The result of this is that the laziness/items cache/printing/iteration/query building/execution are all centralized and only the specific abilities of the RefFeedback are pulled out to the other class.